### PR TITLE
Fix RHEL9 content

### PR DESCRIPTION
**Pull request by ggbecker**
#### Description:

- Make sure jinja if statements apply for RHEL9 whenever possible
  - I kept out of this PR some RHEL9 documentation links that use these if statements because there are no links so far for them.
- Fix an ansible task

#### Rationale:

- Content applies correctly on RHEL9

Partially fix: https://github.com/ComplianceAsCode/content/issues/7874

### CODE REVIEW

File: shared/templates/grub2_bootloader_argument/ansible.template [hunk 1]
```diff
@@ -33,6 +33,7 @@
   ansible.builtin.shell:
     cmd: '/usr/bin/grub2-editenv - list | grep "kernelopts="'
   register: kernelopts
+  ignore_errors: yes
```
Comment (yuumasato) [inline shared/templates/grub2_bootloader_argument/ansible.template:36]: Where should the kernel argument be added when `kernelopts` doesn't exist?

I haven't tested, but it seems the task below will not add the option, and there are no other tasks to handle this case.

**Comment by Mab879** [comment]
Please check the `platform` in the tests. Specifically on `accounts_password_pam_retry`.

File: shared/templates/grub2_bootloader_argument/ansible.template (hunk 1 above)
Comment (ggbecker) [reply shared/templates/grub2_bootloader_argument/ansible.template:36]: That's true. We need to add a new step to set the parameter when kernelopts is not present

Comment (ggbecker) [reply shared/templates/grub2_bootloader_argument/ansible.template:36]: added in 945c02c44588950850beec46fa38d09f2abfd546

**Comment by ggbecker** [comment]
> Please check the `platform` in the tests. Specifically on `accounts_password_pam_retry`.

I believe those are expected to be skipped since some tests are not applicable to Fedora (container we use for testing)

**Comment by Mab879** [comment]
> I believe those are expected to be skipped since some tests are not applicable to Fedora (container we use for testing)

There are specific tests for RHEL8 such as `linux_os/guide/system/accounts/accounts-pam/password_quality/password_quality_pwquality/accounts_password_pam_retry/tests/rhel8_argument_missing.fail.sh` Should we add RHEL9 to these tests?

**Comment by ggbecker** [comment]
> > I believe those are expected to be skipped since some tests are not applicable to Fedora (container we use for testing)
> 
> There are specific tests for RHEL8 such as `linux_os/guide/system/accounts/accounts-pam/password_quality/password_quality_pwquality/accounts_password_pam_retry/tests/rhel8_argument_missing.fail.sh` Should we add RHEL9 to these tests?

We probably should.

**Comment by Mab879** [comment]
Tests pass in a VM with default remediations.
```$ ./test_suite.py rule --libvirt qemu:///system test_ssg_rhel9_nightly --datastream ../build/ssg-rhel9-ds.xml accounts_password_pam_retry
Setting console output to log level INFO
INFO - The base image option has not been specified, choosing libvirt-based test environment.
INFO - Logging into /home/mburket/code/ComplianceAsCode/content/tests/logs/rule-custom-2021-11-16-1226/test_suite.log
WARNING - Script argument_missing.fail.sh is not applicable on given platform
WARNING - Script correct_value.pass.sh is not applicable on given platform
WARNING - Script wrong_value.fail.sh is not applicable on given platform
INFO - xccdf_org.ssgproject.content_rule_accounts_password_pam_retry
INFO - Script rhel8_argument_missing.fail.sh using profile (all) OK
INFO - Script rhel8_correct_value.pass.sh using profile (all) OK
INFO - Script rhel8_wrong_value.fail.sh using profile (all) OK```
```


There seems to be an issue with the Ansible remediations: 


```
$ ./test_suite.py rule --libvirt qemu:///system test_ssg_rhel9_nightly --datastream ../build/ssg-rhel9-ds.xml --remediate-using=ansible accounts_password_pam_retry
Setting console output to log level INFO
INFO - The base image option has not been specified, choosing libvirt-based test environment.
INFO - Logging into /home/mburket/code/ComplianceAsCode/content/tests/logs/rule-custom-2021-11-16-1229/test_suite.log
WARNING - Script argument_missing.fail.sh is not applicable on given platform
WARNING - Script correct_value.pass.sh is not applicable on given platform
WARNING - Script wrong_value.fail.sh is not applicable on given platform
INFO - xccdf_org.ssgproject.content_rule_accounts_password_pam_retry
INFO - Script rhel8_argument_missing.fail.sh using profile (all) OK
ERROR - Ansible playbook remediation run has exited with return code 2 instead of expected 0
ERROR - The remediation failed for rule 'xccdf_org.ssgproject.content_rule_accounts_password_pam_retry'.
INFO - Script rhel8_correct_value.pass.sh using profile (all) OK
INFO - Script rhel8_wrong_value.fail.sh using profile (all) OK
ERROR - Ansible playbook remediation run has exited with return code 2 instead of expected 0
ERROR - The remediation failed for rule 'xccdf_org.ssgproject.content_rule_accounts_password_pam_retry'.
```

**Comment by ggbecker** [comment]
> Tests pass in a VM with default remediations.
> 
> ```
> Setting console output to log level INFO
> INFO - The base image option has not been specified, choosing libvirt-based test environment.
> INFO - Logging into /home/mburket/code/ComplianceAsCode/content/tests/logs/rule-custom-2021-11-16-1226/test_suite.log
> WARNING - Script argument_missing.fail.sh is not applicable on given platform
> WARNING - Script correct_value.pass.sh is not applicable on given platform
> WARNING - Script wrong_value.fail.sh is not applicable on given platform
> INFO - xccdf_org.ssgproject.content_rule_accounts_password_pam_retry
> INFO - Script rhel8_argument_missing.fail.sh using profile (all) OK
> INFO - Script rhel8_correct_value.pass.sh using profile (all) OK
> INFO - Script rhel8_wrong_value.fail.sh using profile (all) OK```
> ```
> 
> There seems to be an issue with the Ansible remediations:
> 
> ```
> $ ./test_suite.py rule --libvirt qemu:///system test_ssg_rhel9_nightly --datastream ../build/ssg-rhel9-ds.xml --remediate-using=ansible accounts_password_pam_retry
> Setting console output to log level INFO
> INFO - The base image option has not been specified, choosing libvirt-based test environment.
> INFO - Logging into /home/mburket/code/ComplianceAsCode/content/tests/logs/rule-custom-2021-11-16-1229/test_suite.log
> WARNING - Script argument_missing.fail.sh is not applicable on given platform
> WARNING - Script correct_value.pass.sh is not applicable on given platform
> WARNING - Script wrong_value.fail.sh is not applicable on given platform
> INFO - xccdf_org.ssgproject.content_rule_accounts_password_pam_retry
> INFO - Script rhel8_argument_missing.fail.sh using profile (all) OK
> ERROR - Ansible playbook remediation run has exited with return code 2 instead of expected 0
> ERROR - The remediation failed for rule 'xccdf_org.ssgproject.content_rule_accounts_password_pam_retry'.
> INFO - Script rhel8_correct_value.pass.sh using profile (all) OK
> INFO - Script rhel8_wrong_value.fail.sh using profile (all) OK
> ERROR - Ansible playbook remediation run has exited with return code 2 instead of expected 0
> ERROR - The remediation failed for rule 'xccdf_org.ssgproject.content_rule_accounts_password_pam_retry'.
> ```

This basically happens because of the first point here: https://github.com/ComplianceAsCode/content/issues/7874#issuecomment-967097673 I think

**Comment by matejak** [comment]
LGTM, I see that not everything is smooth, but personally I would be in favor of merging the PR and address other fixes in a separate one. WDYT?

**Review by Mab879 — APPROVED** [review]

**Pull request merged**